cmd/api/main.go

```diff
@@ -69,29 +92,37 @@ func NewServer(store ItemStore) *server {
 	return s
 }
 
-func (s *server) start(address string) {
-	err := s.router.Start(address)
-	s.router.Logger.Fatal(err)
+func (s *server) start(addr string) error {
+	return s.router.Start(addr)
 }
 
 func (s *server) handleGetAllItems(c echo.Context) error {
-	items := s.store.GetAll()
+	items, err := s.store.GetAll(c.Request().Context())
+	if err != nil {
+		return err
+	}
 
 	return c.JSON(http.StatusOK, map[string]interface{}{
 		"items": items,
 	})
 }
 
 func (s *server) handleGetStories(c echo.Context) error {
-	items := s.store.GetStories()
+	items, err := s.store.GetStories(c.Request().Context())
+	if err != nil {
+		return err
+	}
 
 	return c.JSON(http.StatusOK, map[string]interface{}{
 		"items": items,
 	})
 }
 
 func (s *server) handleGetJobs(c echo.Context) error {
-	items := s.store.GetJobs()
+	items, err := s.store.GetJobs(c.Request().Context())
+	if err != nil {
+		return err
+	}
```

Comment:
I'd maybe move these out into handler function

Reply:
Do you mean something like https://github.com/gymshark/realm-migration-service/blob/06a38636bdb58c352ddae697d8a0dd9ecd69407a/internal/handlers/handler.go#L9 rather than on the `server` struct?

Reply:
Yeah, so that server doesn't get too big